File: 20140126/fizzbuzz.rb
class FizzBuzz
  def initialize(min, max)
    @min = min
    @max = max
  end

  def run
    for num in @min..@max
      fizzbuzz num
    end
  end

  def fizz? num
    return num % 3 == 0
  end
  private :fizz?

  def buzz? num
    return num % 5 == 0
  end
  private :buzz?

  def fizzbuzz num
    if fizz? num and buzz? num
      p "FizzBuzz"
    elsif fizz? num
      p "Fizz"
    elsif buzz? num
      p "Buzz"
    else
      p num
    end
  end
  private :fizzbuzz
end

fb = FizzBuzz.new(1, 100)
fb.run

File: 20140126/fizzbuzz.ml
#!/usr/bin/ocaml

open Printf

let is_fizz t =
 (t mod 3) = 0;;

let is_buzz t =
 (t mod 5) = 0;;

let rec fizzbuzz min max =
 if min <= max then begin
  if is_fizz min && is_buzz min
   then printf "FizzBuzz\n"
  else if is_fizz min
   then printf "Fizz\n"
  else if is_buzz min
   then printf "Buzz\n"
  else
   printf "%d\n" min;
  fizzbuzz (min + 1) max
 end;;

fizzbuzz 1 100;;
